### PR TITLE
[docs] Add note about the project structure in default app 

### PR DESCRIPTION
**Pull request by brentvatne**
# Why

I was chatting with @douglowder about opinions that we express, and project structure (outside of the **app** directory) isn't really one of them. However, he pointed out that people might see this differently given the layout of a default project. So I wanted to add some clarification that people can feel free to do whatever they like. Maybe this is not the best place? Unsure.

image

# How

Add clarification about flexibility of project structure

# Test Plan

Let me know what you think.

### CODE REVIEW

**Comment by amandeepmittal** [comment]
@brent, I think it might be good to explain that here. Here's some context I'd like to share about the on-going work:
- Mark has a PR open on how to make some decisions around organizing a project structure: https://github.com/expo/expo/pull/30847
- Mark's PR (above) explains based on some feedback as to why components cannot be stored inside **app** directory.
- The above PR also aligns with the structure default template's project structure.
- Checking this line (https://github.com/expo/expo/pull/30847/files#diff-04c3f911c12260de6309c4052c7a1a9bb6fcd646a56896bc34dd58ce7f34ec07R145), it says that "Expo recommends sorting components ... by feature" which kind of contradicts "feel free to do..." with the changes in this PR.
- If folks are comfortable modifying their project structure on their own, they are not the right audience for going through this doc
- We added this section from a new user/beginner persona so that they don't have to make decisions when they create a new project, avoid getting overwhelmed, and learn the default template's project structure quickly.

I think once Mark's PR gets merged, here is what we can do:
- Rephrase the context after the "Files" component or use a callout on this page and a link to the "Non-route files" section.

File: docs/pages/get-started/start-developing.mdx [hunk 1]
```diff
@@ -86,7 +86,7 @@ Expo Go is configured by default to automatically reload the app whenever a file
 
 ## File structure
 
-Below, you can get familiar with the default project's file structure:
+Below, you can get familiar with the default project's file structure. This is one of many possible ways to structure your project, and the only enforced convention is usage of the **app** directory for routes with Expo Router. Feel free to rename directories and files and structure your project however you like.
```
Comment (amandeepmittal) [inline docs/pages/get-started/start-developing.mdx:89]:
```suggestion
Below, you can get familiar with the project's default file structure. This is one of many possible ways to structure your project. The only enforced convention is **app** directory for routes with Expo Router. Feel free to rename other directories and files to structure your project however you like.
```

File: docs/pages/get-started/start-developing.mdx [hunk 2]
```diff
@@ -86,7 +86,7 @@ Expo Go is configured by default to automatically reload the app whenever a file
...
 
 <ProjectStructure />
```
Comment (amandeepmittal) [inline docs/pages/get-started/start-developing.mdx:92]:
```suggestion

<br />

> Read more about why [non-route files don't belong in the **app** directory](/router/create-pages/#non-route-files) when using Expo Router.
```

Let's also add this callout that links to the section where non-route files recommended structure is explained.

**Review by amandeepmittal — APPROVED** [review]

File: docs/pages/get-started/start-developing.mdx [hunk 3]
```diff
@@ -86,10 +86,13 @@ Expo Go is configured by default to automatically reload the app whenever a file
...
 ## File structure
 
-Below, you can get familiar with the default project's file structure:
+Below, you can get familiar with the project's default file structure. This is one of many possible ways to structure your project. The only enforced convention is **app** directory for routes with Expo Router. Feel free to rename other directories and files to structure your project however you like.
 
 <ProjectStructure />
 
+<br />
+
+> Read more about why [non-route files don't belong in the **app** directory](/router/create-pages/#non-route-files) when using Expo Router.
```
Comment (brentvatne) [inline docs/pages/get-started/start-developing.mdx:95]:
```suggestion
> Read more about why [non-route files don't belong in the **app** directory](/router/create-pages/#non-route-files) when using Expo Router.

```

File: docs/pages/get-started/start-developing.mdx [hunk 4]
```diff
@@ -86,10 +86,14 @@ Expo Go is configured by default to automatically reload the app whenever a file
 
 ## File structure
 
-Below, you can get familiar with the default project's file structure:
+Below, you can get familiar with the project's default file structure. This is one of many possible ways to structure your project. The only enforced convention is **app** directory for routes with Expo Router. Feel free to rename other directories and files to structure your project however you like.
```
Comment (jonsamp) [inline docs/pages/get-started/start-developing.mdx:89]: Our goal on this doc is to get someone up to speed enough to make a change locally on a project, and this addition doesn't seem to help with that. I'm thinking as someone who doesn't know what Expo Router is, nor are they familiar with us. I suspect they just want to see if they can get something running.

Perhaps we could add a note in the **app** details saying it's a special-cased directory name, specifically for app navigation. That seems helpful for first-time developers to me. It also implicitly hints that the other directories are not special-cased.

**Pull request closed without merging**